src/js/media-chrome-button.ts

```diff
@@ -189,6 +208,12 @@ class MediaChromeButton extends globalThis.HTMLElement {
       } else {
         this.disable();
       }
+    } else if (attrName === 'tooltipposition' && newValue !== oldValue) {
+      // If the tooltip isn't defined, then we could accidentally overwrite
+      // the '.position' property
+      if (this.tooltip && globalThis.customElements.get('media-tooltip')) {
+        this.tooltip.position = newValue;
+      }
```

Comment:
The issue I was seeing that prompted this workaround was that position was not applying ever, if `.position` was attempting to be set before the element was defined. I need to revisit this as the code has changed a bit since but it would be nice not to have to do the check.

I think your last point is covered by the `setupTooltip` method, which runs after `whenDefined` resolves (setup in `connectedCallback`), regardless of if attribute changes have applied yet or not. That method grabs the latest tooltipposition value and sets it on the tooltip so any attribute change misses will be caught at this point. So I _think_ this is fine... but I will do some tests.

Reply:
Ahh right, that makes sense now. The tooltip instance would `hasOwnProperty` `position` so the prototype version would get ignored. (Confirmed in [this writeup](https://nolanlawson.com/2021/08/03/handling-properties-in-custom-element-upgrades/))

This is gonna happen for other custom elements that use tooltips too, so I'm tempted to fix this in Tooltip. You could use the writeup fix and just do the `delete this.position` part since setupTooltip will handle it from there.

Reply:
Very interesting! I implemented the `hasOwnProperty` fix in full and it looks like it works well. I kept the idea of caching the value and setting it after deletion just in case we change our media button implementation one day, and I want to avoid the the tooltip having to have knowledge about how it is being implemented exactly.

Reply:
we use this technique in the theme element too
https://github.com/muxinc/media-chrome/blob/ac8bf0b3d4d59f9875337a626cd1175ab5d0dd43/src/js/media-theme-element.ts#L113-L126

Reply:
I don't know if support is good enough to use it but I noticed this also exists as a replacement for `Object.prototype.hasOwnProperty.call(this, prop)`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn

Reply:
Looks safe enough to me. 👍